src/17FEB0.c

```diff
@@ -40,7 +40,440 @@ s32 calc_item_check_hit(void) {
     return HIT_RESULT_HIT;
 }
 
+// Almost matching besides the parts that are marked as "TODO"
+#ifdef NON_MATCHING
+s32 calc_item_damage_enemy(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Actor* playerActor = battleStatus->playerActor;
+    s32 currentTargetID = battleStatus->currentTargetID;
+    Actor* partnerActor = battleStatus->partnerActor;
+    s32 currentTargetPartID = battleStatus->currentTargetPart;
+    Actor* targetActor;
+    ActorPart* targetActorPart;
+    ActorState* actorState;
+    Evt* attackScript;
+    s32 sp1C = 0;
+    s32 isFireDamage = 0;
+    s32 isElectricDamage = 0;
+    s32 isIceDamage = 0;
+    s8 wasStatusInflicted = 0;
+    s32 currentHP;
+    s32 baseStatusChance;
+    s32 attackDamage;
+    s32 targetActorDefense;
+    s32 dispatchEvent;
+    u32 sp18;
+    s32 phi_a0_2;
+    s32 phi_fp;
+    
+    battleStatus->wasStatusInflicted = 0;
+    battleStatus->lastAttackDamage = 0;
+    battleStatus->attackerActorID = playerActor->actorID;
+    battleStatus->currentTargetID2 = battleStatus->currentTargetID;
+    battleStatus->currentTargetPart2 = battleStatus->currentTargetPart;
+    
+    targetActor = get_actor(currentTargetID);
+    if (targetActor == NULL) {
+        return 0;
+    }
+
+    targetActorPart = get_actor_part(targetActor, currentTargetPartID);
+    currentTargetID &= 0x700;
+    if (targetActorPart == NULL) {
+        PANIC();
+    } 
+
+    targetActor->lastDamageTaken = 0;
+    if (gBattleStatus.flags1 & BS_FLAGS1_80000) {
+        actorState = &playerActor->state;
+    } else {
+        actorState = &partnerActor->state;
+    }
+    phi_a0_2 = 0;
+    if (battleStatus->currentAttackElement & DAMAGE_TYPE_FIRE) {
+        playFX_24(0, actorState->goalPos.x, actorState->goalPos.y, actorState->goalPos.z + 5.0f, 1.0f, 0x18);
+        isFireDamage = TRUE;
+    }
+    if (battleStatus->currentAttackElement & DAMAGE_TYPE_ELECTRIC) {
+        func_80251474(targetActor);
+        isElectricDamage = TRUE;
+    }
+    if (battleStatus->currentAttackElement & DAMAGE_TYPE_WATER) {
+        playFX_5F(0, actorState->goalPos.x, actorState->goalPos.y, actorState->goalPos.z + 5.0f, 1.0f, 0x18);
+    }
+    if (battleStatus->currentAttackElement & DAMAGE_TYPE_ICE) {
+        playFX_30(0, actorState->goalPos.x, actorState->goalPos.y, actorState->goalPos.z + 5.0f);
+        isIceDamage = TRUE;
+    }
+    if ((battleStatus->currentAttackElement & DAMAGE_TYPE_REMOVE_BUFFS) == 0) {
+        if (targetActorPart->eventFlags & ACTOR_EVENT_FLAG_ILLUSORY) {
+            return 6;
+        }
+        if (targetActor->transStatus == 14 || targetActorPart->eventFlags & ACTOR_EVENT_FLAG_800 && !(battleStatus->currentAttackElement & DAMAGE_TYPE_QUAKE)) {
+            return 6;
+        }
+    }
+    if (targetActor->stoneStatus == 12) {
```

Comment:
This and the other status checks (e.g. transStatus) should use the Statuses enum, i.e.
```suggestion
    if (targetActor->stoneStatus == STATUS_STONE) {
```